src/lib/k8s.ts

```diff
@@ -13,30 +11,21 @@ export enum Operation {
 }
 
 /**
- * GenericKind is a generic Kubernetes object that can be used to represent any Kubernetes object
- * that is not explicitly supported by Pepr. This can be used on its own or as a base class for
- * other types. See the examples in `HelloPepr.ts` for more information.
+ * PeprStore for internal use by Pepr. This is used to store arbitrary data in the cluster.
  */
-export class GenericKind {
-  apiVersion?: string;
-  kind?: string;
-  metadata?: V1ObjectMeta;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  [key: string]: any;
+export class PeprStore extends GenericKind {
+  declare data: {
+    [key: string]: string;
+  };
 }
 
-/**
- * GroupVersionKind unambiguously identifies a kind. It doesn't anonymously include GroupVersion
- * to avoid automatic coercion. It doesn't use a GroupVersion to avoid custom marshalling
- **/
-export interface GroupVersionKind {
-  /** The K8s resource kind, e..g "Pod". */
-  readonly kind: string;
-  readonly group: string;
-  readonly version?: string;
-  /** Optional, override the plural name for use in Webhook rules generation */
-  readonly plural?: string;
-}
+export const peprStoreGVK = {
+  kind: "PeprStore",
+  version: "v1",
```

Comment:
I think it's fine as v1 since we have no intention of ever changing, it's effectively a custom ConfigMap to avoid things tracking it for performance reasons.